src/de.rs

```diff
@@ -748,25 +748,22 @@ impl<'de> VariantAccess<'de> for VariantDeserializer<'de> {
     type Error = Error;
 
     fn unit_variant(self) -> Result<(), Error> {
```

Comment:
The changes in this file seem to harm readability IMO. The clippy lint that suggests these should maybe be adjusted...

Reply:
I believe the relevant lint is option_if_let_else. I strongly dislike this lint too. I downgraded it from clippy's `pedantic` category (which is already for low-signal lints) to `nursery` in https://github.com/rust-lang/rust-clippy/pull/7568, meaning it isn't suitable for use.

@gkorland's commit in this PR is titled *"clean clippy nursery warnings"* so it's likely they did something like `-Dclippy::nursery`. That is definitely not an intended use of nursery lints.

```console
$ cargo clippy -- -Dclippy::option_if_let_else

error: use Option::map_or instead of an if let/else
   --> src/de.rs:751:9
    |
751 | /         if let Some(value) = self.value {
752 | |             Deserialize::deserialize(value)
753 | |         } else {
754 | |             Ok(())
755 | |         }
    | |_________^ help: try: `self.value.map_or(Ok(()), |value| Deserialize::deserialize(value))`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#option_if_let_else

error: use Option::map_or_else instead of an if let/else
   --> src/de.rs:762:9
    |
762 | /         if let Some(value) = self.value {
763 | |             seed.deserialize(value)
764 | |         } else {
765 | |             Err(SError::invalid_type(
...   |
768 | |             ))
769 | |         }
    | |_________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#option_if_let_else
help: try
    |
762 ~         self.value.map_or_else(|| Err(SError::invalid_type(
763 +                 Unexpected::UnitVariant,
764 +                 &"newtype variant",
765 +             )), |value| seed.deserialize(value))
    |
```